runelite-client/src/main/java/net/runelite/client/plugins/inventoryviewer/InventoryViewerOverlay.java

```diff
@@ -0,0 +1,100 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.inventoryviewer;
+
+import net.runelite.api.Client;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemContainer;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.ImageComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import javax.inject.Inject;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+
+class InventoryViewerOverlay extends Overlay
+{
+	private static final int INVENTORY_SIZE = 28;
+	private static final int PLACEHOLDER_WIDTH = 36;
+	private static final int PLACEHOLDER_HEIGHT = 32;
+	private static final ImageComponent PLACEHOLDER_IMAGE = new ImageComponent(new BufferedImage(PLACEHOLDER_WIDTH, PLACEHOLDER_HEIGHT, BufferedImage.TYPE_4BYTE_ABGR));
+
+	private final Client client;
+	private final ItemManager itemManager;
+
+	private final PanelComponent panelComponent = new PanelComponent();
+
+	@Inject
+	private InventoryViewerOverlay(Client client, ItemManager itemManager)
+	{
+		setPosition(OverlayPosition.BOTTOM_RIGHT);
+		panelComponent.setWrapping(4);
+		panelComponent.setOrientation(PanelComponent.Orientation.HORIZONTAL);
+		this.itemManager = itemManager;
+		this.client = client;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		final ItemContainer itemContainer = client.getItemContainer(InventoryID.INVENTORY);
+
+		if (itemContainer == null)
+		{
+			return null;
+		}
+
+		panelComponent.getChildren().clear();
+
+		final Item[] items = itemContainer.getItems();
+
+		for (int i = 0; i < INVENTORY_SIZE; i ++)
+		{
+			if (i < items.length)
+			{
+				final BufferedImage image = getImage(items[i]);
+				if (image != null && items[i].getQuantity() > 0)
```

Comment:
Probably checking for quantity before creating image would be better. Also, instead of using items[i] twice just save it and reuse